test/lib/sstable_utils.hh

```diff
@@ -118,6 +118,8 @@ public:
                         return stop_iteration::no;
                     });
                 });
+            }).finally([&] () {
```

Comment:
I see that the same maneuver is already used above because of do_with, but I'll write a comment anyway - please avoid generic captures in lambdas, it's easy to end up with a dangling reference after somebody adds a line here. And `[&ir]` is not that much longer than `[&]` :)

Reply:
also: consider making this call a coroutine - not only `do_with` will get dropped, but also there would be much less indentations involved, making the flow much clearer

Reply:
Another bonus from using a coroutine - it looks like you would no longer need to use smart pointers here, reducing the number of allocations even further. Not to mention that you wouldn't need to use any lambda captures.

Reply:
Oh, just noticed that it's a test, so not really a hot path. Still, a coroutine would be much more concise

Reply:
I think your comment is correct and indeed on master this function is a co routine. It doesn't follow all of the guidelines
that you mentioned but some of them.
The backport will look somehow similar to what you already reviewed but I will do at least the conversion of genric 
capture to a specific one and see if it is satisfactory or not.
10x for the review and thorough explanation :slightly_smiling_face: